clang/lib/Headers/openmp_wrappers/__clang_openmp_device_functions.h

```diff
@@ -36,6 +46,15 @@ extern "C" {
 #pragma omp end declare variant
 
 #ifdef __AMDGCN__
+#pragma omp begin declare variant match(                                       \
+        device = {arch(amdgcn)},                                               \
+            implementation = {extension(disable_implicit_base)})
+
+void *malloc(size_t Size) { return llvm_device_malloc(Size); }
+void free(void *Ptr) { llvm_device_free(Ptr); }
```

Comment:
Unsure how to feel about this, here we always replace `malloc` calls with the one inside the OpenMP device runtime library won't it?

Reply:
You're right. We always replace calls to malloc. I can try your proposal, it's a little unfortunate that we then provide different entry points but that might be the price to pay.